doc/colvars-refman-main.tex

```diff
@@ -2688,14 +2688,20 @@ \subsection{LAMMPS keywords}
 when periodic boundary conditions (PBCs) are defined in the simulation
 and \texttt{distanceZ}'s axis is parallel to a unit cell vector.
 
-The following keywords can be used within periodic components (and are
-illegal elsewhere):
+In addition, a custom \cvscriptonly{or scripted} scalar colvar may be periodic
+depending on its user-defined expression. It will only be treated as such by
+the Colvars module if the period is specified using the \texttt{period} keyword,
+while \texttt{wrapAround} is optional.
+
+The following keywords can be used within periodic components, and within
+custom \cvscriptonly{or scripted} colvars (\ref{sec:colvar_custom_function}
+\cvscriptonly{, \ref{sec::colvar_scripted}}).
 
 \begin{itemize}
 \item %
   \keydef
     {period}{%
-    \texttt{distanceZ}}{%
+    \texttt{distanceZ}, custom colvars}{%
```

Comment:
The keyword period is currently parsed regardless of type: should we disallow it for cases that are not distanceZ or custom/scripted variables?

Reply:
Well, internally they are separate keywords, one at the cvc level and the other at the colvar level. The one at the colvar level is only parsed for scripted/custom colvars.

At the cvc-level, sure we could check the CVC type. Is that known to cvc::init()? This seems like it could be more trouble that it's worth... at worst, those parameters are ignored.

Reply:
Unfortunately not yet, because most derived CVCs still do all initialization in their constructor, which is called after the base class constructor. In the future, they will have their own `init()` functions, and `function_type` (which is set by the derived constructor) at that point should be visible to the base-class `init()`. This is not a huge change, but definitely outside the scope of this PR. Will move ahead.